### PR TITLE
2023 10 25 addressmatchr

### PR DESCRIPTION
**Pull request by AdnanShroufi**
Review for typos, clarity and any omissions.

### CODE REVIEW

**Review by MarkMc1089 — APPROVED** [review]
Nothing major, a couple typos I spotted, and some suggestions to consider.

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+We can use the *source()* function to load in 3 functions, similar to the ones that appear in the {addressMatchR} package, without us having to actually install it.
+As already mentioned, the difference is these functions work with local dataframes rather than remote db tables. 
+
+We will also load in our data for the session, which are the 2 tables we will match against each other.
+'ab' is a sample of addresses from Ordnance Survey (OS) AddressBase (AB).
+'df' is a series of mostly mis-spelt commercial addresses, with some dummy residential addresses included too.
+
+```{r, message=FALSE}
+packages_to_install = c("dplyr", "highcharter", "stringr", "stringdist", "tidytext")
+for(i in packages_to_install){install.packages(i)}
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:66]: Will change install packages.

doubt anyone will knit but can throw in a head() anyway

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+rlang::fn_fmls(calc_match_addresses_df)
+```
+
+## Inspect Table Content
+
+Having had a look at each table metadata, we can have a look at the data itself.
+
+```{r}
+ab
+df
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:99]: added head

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+We can now clean the dummy address data, have a look at it, and compare it against our original dataset.
+
+```{r}
+df
+
+clean_df = df %>%
+  tidy_postcode_df(., POSTCODE) %>% 
+  tidy_single_line_address_df(df = ., col = ADDRESS)
+
+clean_df
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:164]: I won;t change this, can easily compare before and after across output cells

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+```
+
+We will now apply the same cleaning steps to the AB data we will address match the dummy data against.
+
+```{r}
+ab
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:170]: Yes, already cleaned, process is demonstrative

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+
+clean_ab = ab %>%
+  tidy_postcode_df(., AB_POSTCODE) %>% 
+  tidy_single_line_address_df(df = ., col = AB_ADDRESS)
+
+clean_ab
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:176]: not in relation to this df

File: 2023-10-25 addressMatchR/addressMatchR.Rmd [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+```{r}
+output = calc_match_addresses_df(
+    primary_df = clean_df,
+    primary_postcode_col = 'POSTCODE',
+    primary_address_col = 'ADDRESS',
+    lookup_df = clean_ab,
+    lookup_postcode_col = 'AB_POSTCODE',
+    lookup_address_col = 'AB_ADDRESS'
+    )
+```
```
[the remark this answers is not in the document]
Comment (AdnanShroufi) [reply 2023-10-25 addressMatchR/addressMatchR.Rmd:192]: Tried kable but looked ugly, selective column selecting later on

**Comment by AdnanShroufi** [comment]
Changes made, ready to merge

**Pull request merged**